src/types/index.ts

```diff
@@ -19,6 +19,44 @@ type ProcessorTransformationOutput = {
   metadata?: Metadata;
 };
 
+type ProxyDeliveryRequest = {
```

Comment:
should we define ProxyDeliveryRequestCommon with common fields:
type ProxyDeliveryRequestCommon = {
...
};
type ProxyDeliveryRequest = ProxyDeliveryRequestCommon & {
metadata: Metadata;
};
type ProxyDeliveriesRequest = ProxyDeliveryRequestCommon & {
metadata: Metadata[];
};